velox/functions/prestosql/tests/ArrayNormalizeTest.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class ArrayNormalizeTest : public FunctionBaseTest {
+ protected:
+  void testExpr(
+      const VectorPtr& expected,
+      const std::vector<VectorPtr>& input) {
+    auto result =
+        evaluate<ArrayVector>("array_normalize(C0, C1)", makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  // Helper function to calculate p-norm based on its definition.
+  template <typename T>
+  T pNorm(const std::vector<T> values, T p) {
+    T sum = 0;
+    for (const auto& item : values) {
+      sum += pow(abs(item), p);
+    }
+    return pow(sum, 1.0 / p);
+  }
+};
+
+TEST_F(ArrayNormalizeTest, arrayWithElementsZero) {
```

Comment:
Just to be complete (test both float and double) we could write all the tests the following way :
Add in ArrayNormalizeTest

```
template<T>
void testArrayWithElementsEqualZero() {
auto zeroValues = makeArrayVector<T>({{0.0, -0.0, 0.0}});
auto p = makeConstant(2.0, zeroValues->size());
testExpr(zeroValues, {zeroValues, p});
}

```

```
TEST_F(ArrayNormalizeTest, arrayWithElementsZero) {
testArrayWithElementsEqualZero<float>();
testArrayWithElementsEqualZero<double>();
}
```

Reply:
Sounds good, let me update.